### PR TITLE
refactor: use external node type

### PR DESCRIPTION
**Pull request by Victor-C-Zhang**

### CODE REVIEW

File: CMakeLists.txt
Comment (etwest) [inline CMakeLists.txt:175]: Change this before merging

**Review by etwest — APPROVED** [review]
Approve pending change to branch

**Review by etwest — CHANGES_REQUESTED** [review]
Actually, I'm a bit concerned about this. Currently we have two types in the codebase. There's a 64 bit value which we use for the bucket_a values which we call Node (or something) and there's a 32 bit value which we use for node_id_t and bucket_c. I think these changes would need to somehow incorporate both of these types.

**Comment by Victor-C-Zhang** [comment]
> Actually, I'm a bit concerned about this. Currently we have two types in the codebase. There's a 64 bit value which we use for the bucket_a values which we call Node (or something) and there's a 32 bit value which we use for node_id_t and bucket_c. I think these changes would need to somehow incorporate both of these types.

I don’t think `node_id_t` is used in the streaming repo. iirc it’s used to number buffers in the buffer tree only (?)

**Comment by etwest** [comment]
> > Actually, I'm a bit concerned about this. Currently we have two types in the codebase. There's a 64 bit value which we use for the bucket_a values which we call Node (or something) and there's a 32 bit value which we use for node_id_t and bucket_c. I think these changes would need to somehow incorporate both of these types.
> 
> I don’t think `node_id_t` is used in the streaming repo. iirc it’s used to number buffers in the buffer tree only (?)

It's also used by the standalone buffering and we might want to think about expanding its use. 64bit bucket_a only works if we have 32 bit node ids

**Pull request closed without merging**
